### ISSUE TITLE
How to resolve shift/reduce warning conflicts?

### ISSUE THREAD

**Issue by crazydude123** [keosariel/Programming-language-in-python-using-llvmlite#4]
Hi, 
When executing "python3 run.py test/loops.test", say, the terminal warns with the message "WARNING: 30 shift/reduce conflicts". How to resolve this Warning?

Thanks!

**Comment by keosariel** [comment]
Hi @crazydude123 ,

It's probably from the parser, I'd check again and if you can provide anymore info, that'd would be great.

**Comment by crazydude123** [comment]
Hey,

Thanks for your reply.
Are we using o-0 clang optimization here? (Default)

**Comment by 0xosas** [comment]
I didn't set any optimisation. But the llvm-docs has an option for that, with optimisation levels 0-3.

**Comment by keosariel** [comment]
There isn't any optimisation set atm. But the llvm-docs stated that it has an option for that, with optimisation levels 0-3. So you can customize that

**Comment by keosariel** [comment]
Also, concerning the lexer/parser warnings; I intend on changing to a lark parser instead. Hopefully, time permits me.

**Comment by crazydude123** [comment]
Thanks again for responding.
"There isn't any optimisation set atm"
This implies that the optimization level is o-0 right? Is that the default in llvmlite?

**Comment by keosariel** [comment]
I don't think so, essetially the optimization is `False`. Zero (`0`) itself applies some form of optimization and since the optimization is false by default i don't think even the `0` level optimization is applied.

**Comment by keosariel** [comment]
Yeah, sorry I made a mistake on my example.

Also there is a contains function I.e ‘.cs(‘tags’, [“java”])’ returns
results where that tags contains one of the value passed.

And, my apologies for not stating this earlier. The supabase guys have a
more maintained and updated repo here:
https://github.com/supabase-community/supabase-py

Mine was created earlier, cuz there wasn’t a python implementation yet.

So so sorry I didn’t point that out earlier.

Gabriel

**Comment by keosariel** [comment]
Sorry this is a wrong email 😅

Gabriel
